scipy/_lib/_array_api.py

```diff
@@ -645,3 +646,16 @@ def xp_float_to_complex(arr: Array, xp: ModuleType | None = None) -> Array:
         arr = xp.astype(arr, xp.complex128)
 
     return arr
+
+def xp_create_diagonal(x: Array, /, *, offset: int = 0,
+                       xp: ModuleType | None = None) -> Array:
+    if xp is None:
+        xp = array_namespace(x)
+    n = x.shape[0] + abs(offset)
+    diag = xp.zeros(n**2, dtype=x.dtype)
+    if offset >= 0:
+        i = offset
+    else:
+        i = -offset * n
```

Comment:
I know this should be obvious but I was scratching my head for a while wondering how this was working correctly with this negative index. (Solution: index is not negative.)
```suggestion
        i = abs(offset) * n
```